doc/nrf/security/ap_protect.rst

```diff
@@ -197,16 +209,80 @@ This is the default setting in the |NCS|.
 
 You can start debugging the firmware without additional steps needed.
 
-Once you are done debugging, run the following command to enable the access port protection:
+.. _app_approtect_uicr_approtect:
+
+Enabling AP-Protect by locking the UICR.APPROTECT
+=================================================
+
+For the devices in production, you are highly recommended to lock the UICR.APPROTECT register to prevent unauthorized access to the device. The access port protection configured in this way cannot be disabled without erasing the flash.
+
+.. note::
+    This is the only mechanism supported for the nRF52 series and nRF9160 devices that do not have firmware support for AP-Protect.
+
+To lock the UICR.APPROTECT register, follow these steps:
 
 .. code-block:: console
 
    nrfjprog --rbp ALL
 
-This command enables the AP-Protect and resets the device.
+This command enables the AP-Protect (and Secure AP-Protect) and resets the device.
+
+Secure AP-Protect
+=================
+
+With Trusted Firmware-M (TF-M) we have a Secure Processing Environment (SPE) that is isolated from the Non-Secure Processing Environment (NSPE). TF-M is available for the nRF53 and nRF91 Series devices.
+While AP-Protect blocks access to all CPU registers and memories, Secure AP-Protect limits access to the CPU to only non-secure accesses.
+This allows debugging of the NSPE, while SPE debugging is blocked.
+
+The following Kconfig settings are available for nRF91x1 and nRF53 Series devices:
+
+Enabling Secure AP-Protect with :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK`
+==================================================================================
+
+Setting the :kconfig:option:`CONFIG_NRF_SECURE_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware enables the secure access protection mechanism for SoCs of the nRF53 Series.
+
+The setting writes the secure debugger register in SystemInit() in order to lock the secure access port protection in every boot. In addition to this, the UICR.SECUREAPPROTECT register should also be written as instructed in `app_secure_approtect_uicr_approtect`_.
+
+.. note::
+    With multi-image builds, this setting needs to be set for the first image (usually a bootloader). Otherwise the secure AP-Protect will not be sufficient and debugger can be attached to the SPE.
+    With sysbuild, use the SB_CONFIG_SECURE_APPROTECT_LOCK, which sets this for all the images.
```

Comment:
```suggestion

.. note::
    For multi-image builds, this Kconfig option needs to be set for the first image (usually a bootloader).
    Otherwise, the Secure AP-Protect will not be sufficient and the debugger can be attached to the SPE.
    
    When using sysbuild, set the Kcofig option ``SB_CONFIG_SECURE_APPROTECT_LOCK``, which enables Secure AP-Protect for all images.
```